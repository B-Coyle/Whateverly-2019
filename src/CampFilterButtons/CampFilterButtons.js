import React, {Component} from 'react';
import App from '../App/App.js';
import Styles from './CampFilterButtons.scss';

export default class CampFilterButtons extends Component {
  constructor(props) {
    super(props);
    this.state = {
      filters: {
        camping: null,
        elevation: null,
        reservationAvailable: null,
      }
    }
  }

  allowsTents = (props) => {
    let tentCamp = this.props.filteredCampgrounds.filter(campground => campground.camping.includes("Tent"))
    this.props.filteredCampResults(tentCamp);
    return tentCamp;
  }

  allowsCars = (props) => {
    let carCamp = this.props.filteredCampgrounds.filter(campground => campground.camping.includes("Car"))
    this.props.filteredCampResults(carCamp);
    return carCamp;
  }

  allowsRVs = (props) => {
    let rvCamp = this.props.filteredCampgrounds.filter(campground => campground.camping.includes("RV"))
    this.props.filteredCampResults(rvCamp);
    return rvCamp;
  }

  isLess5000 = (props) => {
    let less5000Camp = this.props.filteredCampgrounds.filter(campground => campground['elevation'] <= 4999)
    this.state.updateFilter.push(...less5000Camp);
    return less5000Camp;
  }

  isLess9000 = (props) => {
    let less9000Camp = this.props.filteredCampgrounds.filter(campground => campground['elevation'] >= 5000 && campground['elevation']<= 8999)
    this.state.updateFilter.push(...less9000Camp);
    return less9000Camp;
  }

  isLess11000 = (props) => {
    let less11000Camp = this.props.filteredCampgrounds.filter(campground => campground['elevation'] >= 9000 && campground['elevation']<= 11000)
    this.state.updateFilter.push(...less11000Camp);
    return less11000Camp;
  }

  yesRes = (props) => {
    let yes = this.props.filteredCampgrounds.filter(campground => campground['reservation-available'] === true)
    this.state.updateFilter.push(...yes);
    return yes;
  }

  noRes = (props) => {
    let no = this.props.filteredCampgrounds.filter(campground => campground['reservation-available'] === false)
    this.state.updateFilter.push(...no);
    return no;
  }

  filter = (event) => {
    event.preventDefault();
  }


  render() {
    return (
      <section className='filterContainer'>
        <form>
          <h4 className="formHeader">Please pick one option for each category.</h4>
            <div className='filterSection'>
              <label>Camping Type:</label>
              <div className="filterBtn">
              <p>Tent</p><input type="radio" name="type" value="tent" onChange={this.allowsTents}/>
              </div>
              <div className="filterBtn">
              <p>Car</p><input type="radio" name="type" value="car" onChange={this.allowsCars}/>
              </div>
              <div
              className="filterBtn">
              <p>RV</p><input type="radio" name="type" value="rv" onChange={this.allowsRVs}/>
              </div>
            </div>
            <div className='filterSection'>
              <label>Elevation:</label>
              <div className="filterBtn">
              <p>5000 feet</p><input type="radio" name="elevation" value="less5000" onChange={this.isLess5000}/>
              </div>
              <div className="filterBtn">
              <p>5000-8999 feet</p><input type="radio" name="elevation" value="less9000" onChange={this.isLess9000}/>
              </div>
              <div className="filterBtn">
              <p>9000-11000 feet</p><input type="radio" name="elevation" value="less11000" onChange={this.isLess11000}/>
              </div>
            </div>
            <div className='filterSection'>
              <label>Take a reservation:</label>
              <div className="filterBtn">
                <p>Yes</p><input type="radio" name="reservation" value="Yes" onChange={this.yesRes}/>
                </div>
                <div className="filterBtn">
                <p>No</p><input type="radio" name="reservation" value="No" onChange={this.noRes}/>
                </div>
            </div>
          <input type="submit" className="filterSubmit" onClick={this.filter}/>
          <input type="reset" value="Reset" className="resetSubmit" onClick={this.reset}/>
        </form>  
      </section>
    )
  }
}
