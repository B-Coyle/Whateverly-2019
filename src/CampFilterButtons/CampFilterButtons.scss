.filterContainer {
  margin: 0px 5%;
  display: grid;
  grid-row-start: 2;
  color: white;
  background-color: rgba(63, 39, 46, 0.75);
  font-weight: 600;
  max-height: 900px;
  padding: 10%;
} 

.formHeader {
  margin-top: 0;
  font-weight: 900;
  font-size: 1.2em;
  line-height: 30px;
}

.filterSection {
  margin-top: 25%;
}

.filterBtn {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  margin: 10%;
}

.filterBtn p {
  margin: 0;
}

.filterSubmit, 
  .resetSubmit {
    margin: 15% 6%;
    height: 60px;
    width: 150px;
    font-weight: 900;
  }

  .filterSubmit:hover,
  .resetSubmit:hover {
    color: white;
    background-color: #af5700;
  }