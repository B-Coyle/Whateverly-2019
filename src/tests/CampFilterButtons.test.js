import React from 'react';
import ReactDOM from 'react-dom';
import CampFilterButtons from '../CampFilterButtons/CampFilterButtons.js';
import { shallow } from 'enzyme';

const mockCamping = [
    {
    "name": "Glacier Basin Campground",
    "location": "Estes Park",
    "season": "May-Sept",
    "elevation": 8500,
    "camping": ["Tent", "Car", "RV"],
    "activities": ["Hiking", "Fishing", "Horseback Riding", "Wildlife"],
    "reservation-available": true,
    "nearest-city": null
      },
      {
    "name": "Grand Junction KOA",
    "location": "Grand Junction",
    "season": "All year",
    "elevation": 4600,
    "camping": ["Tent", "Car", "RV"],
    "activities": ["ATVing"],
    "reservation-available": true,
    "nearest-city": null
    }
]

filteredCampgrounds={mockCamping}

describe('CampFilterButtons', () => {
let wrapper

beforeEach(() => {
    wrapper = shallow(
        <CampFilterButtons />
    )
});

// it('should mock the snapshot', () => {
//     expect(wrapper).toMatchSnapshot();
// });

it('should have a default state', () => {
    expect(wrapper.state()).toEqual({
    })
});

})