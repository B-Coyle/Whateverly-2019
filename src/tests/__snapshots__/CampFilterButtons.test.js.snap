// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampFilterButtons should mock the snapshot 1`] = `undefined`;
